File: dart/sum.dart
void main(){
    double i = 0.0;
    double finale = 100000000;
    double step = finale / 10;

    while (i < finale) {
        i = i+1;
        if (i % step == 1) {
            print("\r ${i/finale*100}%");
        }
    }
    print("\n ${i.toString()} end\n");
}

File: perl/sum.pl
use strict;
use warnings;

$| = 1;

my $i = 0;
my $final = 1_000_000_000;
my $step = $final/10;
while ($i < $final) {
    $i++;
    if ($i % $step == 1) {
        printf("\r%.1f%%", $i/$final*100);
    }
}
print "\n",$i, ' end';
